### ISSUE TITLE
Error while trying to download (code:0): 

### ISSUE THREAD

**Issue by gvdkooij** [getgrav/grav-plugin-admin#1626]
When I try to upgrade  Admin Panel (in the Admin Panel), from v1.8.17 to v1.8.19 I get the following error:

Error while trying to download (code:0): 
https://getgrav.org/download/plugins/admin/1.8.19 Message: URL using bad/illegal format or missing URL

With SSH and GPM command it works fine.

![40616](https://user-images.githubusercontent.com/46501812/53599292-2225f180-3ba7-11e9-81e2-a21734b094c1.jpg)

**Comment by rhukster** [comment]
Can you try switching from Curl to Fopen in the system config or vice versa?

**Comment by ninjaring** [comment]
same case here. plus, i can't download any plugin

**Comment by ninjaring** [comment]
i'm using shared hosting

**Comment by w00fz** [comment]
Can you check your php is the same for both cli and web server?

**Comment by w00fz** [comment]
Also can you let us know if this is the first time you are trying an update or if the issue just started showing up but you previously had successful updates?

**Comment by buitermark** [comment]
I stubled upon this same problem also.
I think i now know a bit more about it.. Using it on a shared hosting also.

When i go into Configuration, and under System go to Advanced change External Fetch Method to "fopen" AND Remote Verify Peer (SSL) to "No" (although running with an cert from LetsEncrypt)
i'am suddenly able to download and install the updates again.. 

Maybe this works for others also..

**Comment by PixelPW** [comment]
> When i go into Configuration, and under System go to Advanced change External Fetch Method to "fopen" AND Remote Verify Peer (SSL) to "No" (although running with an cert from LetsEncrypt)
> i'am suddenly able to download and install the updates again..

I had the same problem. This worked for me.

**Comment by mahagr** [comment]
You have likely too old version of SSL root certificates in your server. Those can usually be updated by upgrading the server software.

**Comment by buitermark** [comment]
> You have likely too old version of SSL root certificates in your server. Those can usually be updated by upgrading the server software.
> 
> PS. this is different from your own server SSL certificate.

The problem mentioned here is about **Shared Hosting** so, the only way of upgrading the SSL root certificate is asking the shared hosting provider, or move to another povider....
A user can't do anything else about the root cert of a shared host..

**Comment by mahagr** [comment]
What you just said raises a red flag for me. I would contact the hosting about the issue and if there's no response, move to somewhere else. There is no point of staying in a host which does not keep the servers up to date. :)

**Comment by ekenberg** [comment]
Is there a way to recreate this error using curl from the commandline (ssh) on the server? This would help a lot when dealing with hosting providers to show the error and make it easier for them to verify that a root certificate upgrade really solves the problem.

**Comment by mahagr** [comment]
Yes, just get the data from your browser and convert it to be compatible to CURL -- I am pretty sure there are also tools for that. The only caveat is that you need to be logged in + have the nonce token, which means that the request needs to be changed slightly.

Though that said, It should not be hard to just give admin user and go through the steps to reproduce the issue. It'll likely take less time from everyone.

**Comment by ekenberg** [comment]
Thank you @mahagr, but I think there's a misunderstanding. You talk about using curl to access grav admin pages to recreate the problem? I mean something else:

Since changing from curl to fopen in grav system config solves it, there should be a curl call from grav going wrong internally? It's that call I want to extract and recreate manually on the commandline.

**Comment by mahagr** [comment]
Oh, I got it this time -- using `curl` makes it fail and `fopen` solves the issue. 

Basically, people are saying that changing setting `Remote Verify Peer (SSL)` to `No` fixes the issue, which means that the SSL certificates installed to the server are old.

**Comment by IfThenElvis** [comment]
I am on shared hosting and the issue was a blocked outgoing connection that I could whitelist in my webhost's control panel,

**Comment by ulittle** [comment]
> I stubled upon this same problem also.
> I think i now know a bit more about it.. Using it on a shared hosting also.
> 
> When i go into Configuration, and under System go to Advanced change External Fetch Method to "fopen" AND Remote Verify Peer (SSL) to "No" (although running with an cert from LetsEncrypt)
> i'am suddenly able to download and install the updates again..
> 
> Maybe this works for others also.

I updated to version 1.6.22 using the above steps - thank you.
Note: **External Fetch Method** on my version is **Remote Fetch Method**
![Remote Fetch Method](https://user-images.githubusercontent.com/8032961/76142142-f2190480-60be-11ea-9772-abc267987289.png)

**Comment by ulab** [comment]
I am running into this issue myself (shared hosting, but I am the admin. Debian 9.12, packages are up to date).

Changing the Fetch Method to fopen and Remote Verify Peer to No doesn't help. I still get an Invalid AJAX response. 

I can manually follow the redirects with `curl -v` and download the file in the end. So I thought I change the Fetch Method to cURL, same issue.

**Comment by ddoquang** [comment]
I ran into this issue on a self-hosted CentOS 8 server.  SELinux was blocking network connections for the httpd process.

Connect to the server by ssh and run the following (you need to be admin):
`sudo sestatus -b |grep httpd_can_network_connect`
Default is "off".

Set it to "on" 
`sudo setsebool -P httpd_can_network_connect 1`

Once that is done the issue should be resolved.

**Comment by ezchile** [comment]
Same issue here. Changed system to fopen and 'Remote Verify Peer (SSL)' to no.. No change, still errors out.  

Shared MediaTemple Grid hosting..

**Comment by mahagr** [comment]
Grav 1.7 has improvements (using a Symfony library) on downloading updates. Can you try (in a test site) if they fix the issue?

**Comment by ezchile** [comment]
> Grav 1.7 has improvements (using a Symfony library) on downloading updates. Can you try (in a test site) if they fix the issue?

It's fixed downloading updates, though still shows 'fetch failed' for things like 'Purge Old Cache' ('Fetch Failed:
Purged 1 old cache folders...{"status":"success","message":null}')

**Comment by mahagr** [comment]
@ezchile Can you please create a new issue about this?

**Comment by ezchile** [comment]
@mahagr Sure. I just have:  [https://github.com/getgrav/grav-plugin-admin/issues/1935](https://github.com/getgrav/grav-plugin-admin/issues/1935)

**Comment by mahagr** [comment]
Thanks, it's easier to follow an open issue. :)
